### PR TITLE
shift model and API

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR adds the model and API for the scanning shifts.

### CODE REVIEW

**Comment by mcoughlin** [comment]
@guynir42 @profjsb I removed any front-end stuff from this one, which hopefully means we can still check interactions enough to decide if it is alright.

File: alembic/versions/5ab82f23e15c_add_shift.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+Create Date: 2022-02-01 05:24:23.927168
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '5ab82f23e15c'
+down_revision = '82f5a89fbe59'
```
Comment (profjsb) [inline alembic/versions/5ab82f23e15c_add_shift.py:14]: will need to update the down revision given that there are new migrations.

File: skyportal/handlers/api/shift.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                    mode="read",
+                    options=[joinedload(Shift.group).joinedload(Group.users)],
+                )
+                .order_by(Shift.start_date.asc())
+                .all()
+            )
+        self.verify_and_commit()
+        return self.success(data=shifts)
+
+    @permissions(["Upload data"])
```
Comment (profjsb) [inline skyportal/handlers/api/shift.py:108]: should be a permissions to "Manage shifts"

File: skyportal/handlers/api/shift.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+from ..base import BaseHandler
+from ...models import (
+    DBSession,
+    Group,
+    Shift,
+)
+
+
+class ShiftHandler(BaseHandler):
+    @permissions(["Upload data"])
```
Comment (profjsb) [inline skyportal/handlers/api/shift.py:13]: Should be a permission to "Manage shifts"

File: skyportal/tests/api/test_shifts.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+from skyportal.tests import api
+
+
+def test_shift(public_group, super_admin_token, super_admin_user):
```
Comment (profjsb) [inline skyportal/tests/api/test_shifts.py:4]: Should be testing the deletion of a shift both with and without proper permissions.

File: skyportal/model_util.py [hunk 5]
```diff
@@ -10,6 +10,7 @@
     'Manage users',
     'Manage sources',
     'Manage groups',
+    'Manage shifts',
```
Comment (profjsb) [inline skyportal/model_util.py:13]: Also add "Manage shifts" to the "Group admin" role below (line 24). I think that's the default behavior you want...

**Review by profjsb — CHANGES_REQUESTED** [review]

File: skyportal/model_util.py (hunk 5 above)
Comment (mcoughlin) [reply skyportal/model_util.py:13]: added

File: skyportal/tests/api/test_shifts.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+    request_data = {
+        'group_id': public_group.id,
+        'start_date': '3021-02-27T00:00:00',
+        'end_date': '3021-07-20T00:00:00',
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/tests/api/test_shifts.py:10]: added

File: skyportal/tests/api/test_shifts.py (hunk 4 above)
Comment (mcoughlin) [reply skyportal/tests/api/test_shifts.py]: added

File: skyportal/handlers/api/shift.py (hunk 3 above)
Comment (mcoughlin) [reply skyportal/handlers/api/shift.py]: added

File: skyportal/handlers/api/shift.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/handlers/api/shift.py]: added

File: alembic/versions/5ab82f23e15c_add_shift.py (hunk 1 above)
Comment (mcoughlin) [reply alembic/versions/5ab82f23e15c_add_shift.py:14]: The migration script is now complaining about bandpasses... although I expect this must be not related to this PR

**Pull request closed without merging**